### PR TITLE
Mozc, Protobuf, GYP, MARISA, OpenCC, UTF8-CPP etc. fixes

### PR DESCRIPTION
**Pull request by Arfrever**

### CODE REVIEW

**Comment by Arfrever** [comment]
CJK team: floppym is no longer able to proxy my commits, so I would like to transfer co-maintainership of these packages to CJK:
* app-i18n/tagainijisho (Japanese dictionary)
* dev-cpp/glog (dependency of app-i18n/librime which is co-maintained by Arfrever + CJK)
* dev-libs/protobuf (dependency of app-i18n/mozc which is co-maintained by Arfrever + CJK)
* dev-libs/utfcpp (dependency of app-i18n/librime which is co-maintained by Arfrever + CJK)
* dev-util/gyp (dependency of app-i18n/mozc which is co-maintained by Arfrever + CJK)

**Comment by dlan17** [comment]
hi, Arfrever, I can help on this
1) I will try to review and test locally
2) may proxy commit after that..

note: I'd more than happy if someone else from @gentoo/cjk wiling to help..

File: app-i18n/mozc/files/mozc-2.23.2815.102-protobuf-3.18.patch [hunk 1]
```diff
@@ -0,0 +1,11 @@
+--- /src/dictionary/user_dictionary_storage.cc
++++ /src/dictionary/user_dictionary_storage.cc
```
Comment (SoapGentoo) [inline app-i18n/mozc/files/mozc-2.23.2815.102-protobuf-3.18.patch:2]: please use the git patch style, like the rest of us (this applies to all other patches too)

File: app-i18n/opencc/opencc-1.1.2.ebuild [hunk 2]
```diff
@@ -1,8 +1,8 @@
 # Copyright 2010-2022 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI="7"
-PYTHON_COMPAT=( python3_{8,9} )
+EAPI="8"
+PYTHON_COMPAT=(python{3_8,3_9,3_10})
```
Comment (SoapGentoo) [inline app-i18n/opencc/opencc-1.1.2.ebuild:5]:
```suggestion
PYTHON_COMPAT=( python3_{8..10} )
```

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

**Review by dlan17 — COMMENTED** [review]
this commit actually trigger a warning for me, any reference that don't need this?
 * QA Notice: DISTUTILS_USE_SETUPTOOLS is not used when DISTUTILS_OPTIONAL
 * is enabled.

**Comment by dlan17** [comment]
general comments here.

1) I'd also like to see EAPI bump to 8 if possible..

2) I don't really like numerous live ebuilds here, it just make maintenance difficult.
there are even patches applied to app-i18n/mozc-9999 which seems wrong to me.

3) I'd also hope we at least could squash patches for mozc into tarball, then move out of portage tree, so users don't have to carry them, something like we do in app-emulation/xen. or even better pushing those patches to upstream. still, it's ok to not address this in this PR, but try to improve later ..

File: app-i18n/mozc/files/mozc-2.23.2815.102-protobuf-3.18.patch (hunk 1 above)
Comment (dlan17) [reply app-i18n/mozc/files/mozc-2.23.2815.102-protobuf-3.18.patch:2]: +1, I'd also appreciate if you could provide commit messages and keep the author/signed-off information when doing git-format-patch

**Comment by Arfrever** [comment]
I added additional updates to `EAPI="8"` for 3 packages. I will add more in future pull-request.

That fix for dev-libs/marisa present in this pull-request fixes that `QA Notice: DISTUTILS_USE_SETUPTOOLS is not used when DISTUTILS_OPTIONAL is enabled.` (i.e. fixes https://bugs.gentoo.org/800290) which is no longer reproducible anyway, because whole QA check was deleted on 2022-01-16: f344326fa3c3653d85acf12bf053bfd98f354c56

File: app-i18n/opencc/opencc-1.1.2.ebuild [hunk 3]
```diff
@@ -1,8 +1,8 @@
 # Copyright 2010-2022 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI="7"
-PYTHON_COMPAT=( python3_{8,9} )
+EAPI="8"
+PYTHON_COMPAT=(python3_{8..10})
```
Comment (SoapGentoo) [inline app-i18n/opencc/opencc-1.1.2.ebuild:5]:
```suggestion
PYTHON_COMPAT=( python3_{8..10} )
```
do not remove spacing

**Review by SoapGentoo — CHANGES_REQUESTED** [review]
please also fix the patches, as I said before

File: app-i18n/opencc/opencc-1.1.2.ebuild (hunk 3 above)
Comment (Arfrever) [reply app-i18n/opencc/opencc-1.1.2.ebuild:5]: These spaces are unnecessary.
They are also not included in canonical display of array variable using `declare -p`:
```
$ myarray=( a b c )
$ declare -p myarray
declare -a myarray=([0]="a" [1]="b" [2]="c")
```

There are different tastes, and there is nothing wrong with it.
Some people prefer:
```
if something; then
```
And other people prefer:
```
if something ; then
```
Some people prefer:
```
EAPI=8
```
And other people prefer:
```
EAPI="8"
```

**Comment by Arfrever** [comment]
[Devmanual](https://devmanual.gentoo.org/ebuild-writing/misc-files/patches/index.html) contains examples of various patches, and last 3 examples are of clean patches without any git-specific elements.
(And please do not think about changing Devmanual to suit your antidiversity views.)

File: app-i18n/opencc/opencc-1.1.2.ebuild (hunk 3 above)
Comment (SoapGentoo) [reply app-i18n/opencc/opencc-1.1.2.ebuild:5]: Ok, I'll put it plain and simple: either you change it, or this won't go in. It's that simple.

**Comment by SoapGentoo** [comment]
1. do not add a `*/*` commit, squash the "cosmetic" changes into the right commit
2. do not add full-stops after the first line. These aren't sentences, and noone does it.
3.
```
--- a/src/dictionary/user_dictionary_storage.cc
+++ b/src/dictionary/user_dictionary_storage.cc
```
format as I said before.

File: app-i18n/mozc/mozc-2.23.2815.102.ebuild [hunk 4]
```diff
@@ -1,8 +1,8 @@
-# Copyright 2010-2021 Gentoo Authors
+# Copyright 2010-2022 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI="7"
-PYTHON_COMPAT=(python{3_7,3_8,3_9})
+EAPI="8"
+PYTHON_COMPAT=(python3_{8..10})
```
Comment (SoapGentoo) [inline app-i18n/mozc/mozc-2.23.2815.102.ebuild:5]:
```suggestion
PYTHON_COMPAT=( python3_{8..10} )
```

File: app-i18n/mozc/mozc-2.26.4220_p20201212102434_p20201219202429.ebuild [hunk 5]
```diff
@@ -1,8 +1,8 @@
-# Copyright 2010-2021 Gentoo Authors
+# Copyright 2010-2022 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI="7"
-PYTHON_COMPAT=(python{3_7,3_8,3_9})
+EAPI="8"
+PYTHON_COMPAT=(python3_{8..10})
```
Comment (SoapGentoo) [inline app-i18n/mozc/mozc-2.26.4220_p20201212102434_p20201219202429.ebuild:5]:
```suggestion
PYTHON_COMPAT=( python3_{8..10} )
```

File: app-i18n/opencc/opencc-1.1.3.ebuild [hunk 6]
```diff
@@ -0,0 +1,69 @@
+# Copyright 2010-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="8"
+PYTHON_COMPAT=(python3_{8..10})
```
Comment (SoapGentoo) [inline app-i18n/opencc/opencc-1.1.3.ebuild:5]:
```suggestion
PYTHON_COMPAT=( python3_{8..10} )
```

File: app-i18n/opencc/opencc-9999.ebuild [hunk 7]
```diff
@@ -1,8 +1,8 @@
-# Copyright 2010-2021 Gentoo Authors
+# Copyright 2010-2022 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI="7"
-PYTHON_COMPAT=(python{3_7,3_8,3_9})
+EAPI="8"
+PYTHON_COMPAT=(python3_{8..10})
```
Comment (SoapGentoo) [inline app-i18n/opencc/opencc-9999.ebuild:5]:
```suggestion
PYTHON_COMPAT=( python3_{8..10} )
```

File: dev-libs/marisa/marisa-9999.ebuild [hunk 8]
```diff
@@ -1,10 +1,9 @@
-# Copyright 2014-2020 Gentoo Authors
+# Copyright 2014-2022 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI="7"
-PYTHON_COMPAT=(python{3_7,3_8,3_9})
+EAPI="8"
+PYTHON_COMPAT=(python3_{8..10})
```
Comment (SoapGentoo) [inline dev-libs/marisa/marisa-9999.ebuild:5]:
```suggestion
PYTHON_COMPAT=( python3_{8..10} )
```

File: dev-libs/protobuf/files/protobuf-3.19.0-disable_no-warning-test.patch [hunk 9]
```diff
@@ -0,0 +1,19 @@
+Disable no-warning-test which is compiled with -Werror option and whose only purpose is checking if compilation results in any warnings.
+
+--- /src/Makefile.am
++++ /src/Makefile.am
```
Comment (SoapGentoo) [inline dev-libs/protobuf/files/protobuf-3.19.0-disable_no-warning-test.patch:4]: fix

File: dev-libs/utfcpp/utfcpp-3.2.1.ebuild [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+
+LICENSE="Boost-1.0"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc ~ppc64 ~sparc ~x86"
+IUSE="test"
+RESTRICT="!test? ( test )"
+
+BDEPEND=""
+DEPEND=""
+RDEPEND=""
```
Comment (SoapGentoo) [inline dev-libs/utfcpp/utfcpp-3.2.1.ebuild:35]: remove all of these

File: dev-python/protobuf-python/files/protobuf-python-3.19.0-google.protobuf.pyext._message.PyUnknownFieldRef.patch [hunk 11]
```diff
@@ -0,0 +1,25 @@
+Prevent integer overflow for unknown fields.
+
+https://github.com/protocolbuffers/protobuf/issues/6205
+https://github.com/protocolbuffers/protobuf/pull/7016
+https://github.com/protocolbuffers/protobuf/commit/5100be2b7746391c2724e2793e1428c36b63c98b
+
+--- /python/google/protobuf/pyext/unknown_fields.cc
++++ /python/google/protobuf/pyext/unknown_fields.cc
```
Comment (SoapGentoo) [inline dev-python/protobuf-python/files/protobuf-python-3.19.0-google.protobuf.pyext._message.PyUnknownFieldRef.patch:8]: git format

File: dev-util/gyp/gyp-20200512145953.ebuild [hunk 12]
```diff
@@ -1,10 +1,9 @@
-# Copyright 2017-2020 Gentoo Authors
+# Copyright 2017-2022 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI="7"
-PYTHON_COMPAT=(python{3_7,3_8,3_9})
+EAPI="8"
+PYTHON_COMPAT=(python3_{8..10})
```
Comment (SoapGentoo) [inline dev-util/gyp/gyp-20200512145953.ebuild:5]:
```suggestion
PYTHON_COMPAT=( python3_{8..10} )
```

File: dev-util/gyp/gyp-99999999999999.ebuild [hunk 13]
```diff
@@ -1,10 +1,9 @@
-# Copyright 2017-2020 Gentoo Authors
+# Copyright 2017-2022 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI="7"
-PYTHON_COMPAT=(python{3_7,3_8,3_9})
+EAPI="8"
+PYTHON_COMPAT=(python3_{8..10})
```
Comment (SoapGentoo) [inline dev-util/gyp/gyp-99999999999999.ebuild:5]:
```suggestion
PYTHON_COMPAT=( python3_{8..10} )
```

**Review by SoapGentoo — CHANGES_REQUESTED** [review]
1. do not add a `*/*` commit, squash the "cosmetic" changes into the right commit
2. do not add full-stops after the first line in the commit message. These aren't sentences, and noone does it.
3.
```
--- a/src/dictionary/user_dictionary_storage.cc
+++ b/src/dictionary/user_dictionary_storage.cc
```
format as I said before.

**Comment by dlan17** [comment]
hi guys, 
 let's be more practical & productive about this (less bikeshedding), I've just went ahead and fixed most of those style issues.
1) I've format patches using "git-format-patch style", which I think it's will become more maintainable. (leave out mozc's patch, which I'd hope to improve later - maybe next PR/version bump)
2) adjust PYTHON_COMPAT (which isn't difficult..)
3) leave full-stop issue which doesn't really hurt me

**Pull request closed without merging**